examples/waldo/core/src/image.rs

```diff
@@ -185,8 +185,10 @@ mod zkvm {
                     let chunk = self
                         .chunks
                         .get(usize::try_from(y * self.width_chunks + x).unwrap());
-                    self.cache
-                        .insert((x, y), RgbImage::from(chunk.into()).into());
+                    self.cache.insert(
+                        (x, y),
+                        RgbImage::from(<ImageChunk as Into<RgbImage>>::into(chunk)).into(),
```

Comment:
I think this could be simpler
```suggestion
                        RgbImage::from(RgbImage::from(chunk)).into(),
```

Reply:
@nvzqz I tried this suggestion and ended up with this type error:
```
image_crop:    Compiling waldo-core v0.1.0 (/Users/erik/risc0/examples/waldo/core)ldo-methods(build), rustls-webpki...
image_crop: error[E0277]: the trait bound `ImageBuffer<Rgb<u8>, Vec<u8>>: From<ImageChunk>` is not satisfied-webpki   
image_crop:    --> /Users/erik/risc0/examples/waldo/core/src/image.rs:189:56
image_crop:     |
image_crop: 189 |                         .insert((x, y), RgbImage::from(RgbImage::from(chunk)).into());
image_crop:     |                                                        ^^^^^^^^ the trait `From<ImageChunk>` is not implemented for `ImageBuffer<Rgb<u8>, Vec<u8>>`
image_crop:     |
image_crop:     = help: the trait `From<DynamicImage>` is implemented for `ImageBuffer<Rgb<u8>, Vec<u8>>`
image_crop:     = help: for that trait implementation, expected `DynamicImage`, found `ImageChunk`
image_crop: 
image_crop: For more information about this error, try `rustc --explain E0277`.
image_crop: error: could not compile `waldo-core` (lib) due to 1 previous error
error: failed to run custom build command for `waldo-methods v0.1.0 (/Users/erik/risc0/examples/waldo/methods)`
```
rustc seems to fail to infer the type synonyms..? I'm going to revert to the previous code for now but let me know if you have a different suggestion

Reply:
It seems the issue is we need to do `impl From<ImageChunk> for RgbImage` instead of the current `Into` implementation